### ISSUE TITLE
MANGO: A Mask Attention Guided One-Stage Scene Text Spotter

### ISSUE THREAD

**Issue by AtsukiOsanai** [AtsukiOsanai/cv_survey#98]
# MANGO: A Mask Attention Guided One-Stage Scene Text Spotter

## Information
- Authors: Liang Qiao, Ying Chen+
- Organization: Hikvision Research Institute
- Paper: https://arxiv.org/abs/2012.04350
- Code: https://github.com/hikopensource/davar-lab-ocr
- Conference/Journal: AAAI2021

### サマリ図表
<img width="758" alt="スクリーンショット 2023-01-10 6 37 20" src="https://user-images.githubusercontent.com/42481903/211413251-15b7fed1-3002-458d-888f-64a66ace8f3b.png">

### どんな論文か？
Two-stage(detect & recognize)のText Spotterではdetectionの精度がrecognitionのlimitationとなるため、本論文ではOne-StageのText Spotterを提案する。提案手法はInstance Mask Attention(IMA), Character Mask Attention(CMA)モジュールを有することで、画像からCharacter-levelのfeature sequenceを得ることができる。これをRecognizerの基本パターン(RNN + Linear)に通して文字系列をE2Eに推定可能。また、Centerline segmentationをsub taskとして解いておき、テキストインスタンスの区別のために利用する。

### 新規性
- 2種類のMask推定を解したE2E One-stage Text Spotterの提案
- Centerline segmentationを元にした、character-weighted votingによるinference方法
  - 1つのtext instanceに対してinstance maskが複数でるため、それらの結果をrecognitionのscoreで重み付け平均し最終出力を決定する

### 結果
- E2EのText/Word spottingでSOTA
- VisualizationではChar-levelでの検出もできている

<img width="756" alt="スクリーンショット 2023-01-10 6 34 50" src="https://user-images.githubusercontent.com/42481903/211413040-16167054-8c50-4e8c-a67b-d4b42c3f1393.png">

<img width="742" alt="スクリーンショット 2023-01-10 6 35 07" src="https://user-images.githubusercontent.com/42481903/211413176-9f1493ee-868f-4c01-9793-5c17b63bd6e0.png">

### その他（なぜ通ったか？など）
- pre-trainingに必要なsupervisionはCenteline segmentation, instance mask, character mask, recognition(text sequence)なので結構重い。SynthTextを使用。
- fine-tuning stageではinstance levelのtext, polygon annotationが必要。Centerline segとrecognitionのlossのみを考慮するようである
- bbox annotationのみよりはpolygonの方が好ましい（Table5）
- project page: https://davar-lab.github.io/publication/aaai2021_mango.html
- DAVAR-Lab：https://davar-lab.github.io/publication.html
